File: matlab/trackers/LSK/run_LSK.m
function results=run_LSK(seq, res_path, bSaveImage)
%LSK is based on matlab

close all;

outputPath = './results';
configPath = './config/';

% seq.len = 100;
% seq.endFrame = seq.startFrame + seq.len - 1;

r = seq.init_rect;

name_sub_gt=[configPath seq.name '_gt.txt'];
dlmwrite(name_sub_gt,r);

xmlName = [configPath seq.name '.xml'];
fid=fopen(xmlName,'w');

sizeC=[30,30];

ratio = (sizeC(1)*sizeC(2))/(r(3)*r(4));
if ratio>=1
    ratio = 1;
else
    ratio = ceil(ratio*10)/10;
end

fprintf(fid,'<xml>\n');
fprintf(fid,'\t<properties>\n');
fprintf(fid,'\t\t<!-- resize the image to specified scale for tracking  -->\n');
fprintf(fid,'\t\t<imgScale>%.2f</imgScale>\n',ratio);
fprintf(fid,'\t\t<!-- the patch size -->\n');
fprintf(fid,'\t\t<patchSize>5</patchSize>\n');
fprintf(fid,'\t\t<!-- dictionary size (percentage) -->\n');
fprintf(fid,'\t\t<dictionarySize>0.15</dictionarySize>\n');
fprintf(fid,'\t\t<!-- the sparsity parameter K -->\n');
fprintf(fid,'\t\t<sparsityK>3</sparsityK>\n');
fprintf(fid,'\t</properties>\n');

fprintf(fid,'\t<sequence>\n');
fprintf(fid,'\t\t<name>%s</name>\n',seq.name);
fprintf(fid,'\t\t<gtFile>%s</gtFile>\n',name_sub_gt);
fprintf(fid,'\t\t<imgFolder>%s</imgFolder>\n',seq.path);
fprintf(fid,'\t\t<imgIdFormat>%%0%dd</imgIdFormat>\n',seq.nz);
fprintf(fid,'\t\t<imgExt>%s</imgExt>\n',seq.ext);
fprintf(fid,'\t\t<startFrame>%d</startFrame>\n',seq.startFrame);
fprintf(fid,'\t\t<endFrame>%d</endFrame>\n',seq.endFrame);
fprintf(fid,'\t\t<writeImage>%d</writeImage>\n', 0);
fprintf(fid,'\t\t<showResult>%d</showResult>\n', 0);
%fprintf(fid,'\t\t<outputFolder>./out_%s_%d</outputFolder>\n', s.name, idx);
fprintf(fid,'\t\t<outputFolder>%s</outputFolder>\n',outputPath);
fprintf(fid,'\t</sequence>\n');

fprintf(fid,'</xml>\n');
fclose(fid);

if ~exist(outputPath,'dir')
    mkdir(outputPath)
end

tic
command = ['spt64.exe ' xmlName];
dos(command);
duration=toc;

pause(1)

results.res = dlmread([outputPath '/' seq.name '.txt']);

results.type='rect';
results.fps=seq.len/duration;

% results.fps = dlmread([outputPath seq.name '_MIL_FPS.txt']);

File: scripts/bscripts/run_LSK.py
import subprocess
import time
import math
import numpy as np
from config import *

def run_LSK(seq, rp, bSaveImage):
    x = seq.init_rect[0] - 1
    y = seq.init_rect[1] - 1
    w = seq.init_rect[2]
    h = seq.init_rect[3]

    path = './results/'
    config = './config/'

    if not os.path.exists(path):
        os.makedirs(path)
    if not os.path.exists(config):
        os.makedirs(config)

    name_sub_gt = config + seq.name + '_gt.txt'
    gtfile = open(name_sub_gt, 'w')
    gtfile.write(', '.join(map(str, seq.init_rect)))
    gtfile.close()

    name_xml = config + seq.name + '.xml'
    xmlfile = open(name_xml, 'w')

    sizeC = [30.0, 30.0]
    ratio = (sizeC[0]*sizeC[1])/(seq.init_rect[2]*seq.init_rect[3])
    if ratio >= 1:
        ratio = 1
    else:
        ratio = math.ceil(ratio*10)/10.0

    xmlstr = \
        '<xml>\n' + \
        '\t<properties>\n' + \
        '\t\t<!-- resize the image to specified scale for tracking -->\n' + \
        '\t\t<imgScale>{0:.2f}</imgScale>\n'.format(ratio) + \
        '\t\t<!-- the patch size -->\n' + \
        '\t\t<patchSize>5</patchSize>\n' + \
        '\t\t<!-- dictionary size (percentage) -->\n' + \
        '\t\t<dictionarySize>0.15</dictionarySize>\n' + \
        '\t\t<!-- the sparsity parameter K -->\n' + \
        '\t\t<sparistyK>3</sparistyK>\n' + \
        '\t</properties>\n' + \
        \
        '\t<sequence>\n' + \
        '\t\t<name>{0}</name>\n'.format(seq.name) + \
        '\t\t<gtFile>{0}</gtFile>\n'.format(name_sub_gt) + \
        '\t\t<imgFolder>{0}</imgFolder>\n'.format(seq.path) + \
        '\t\t<imgIdFormat>%{0:02d}d</imgIdFormat>\n'.format(seq.nz) + \
        '\t\t<imgExt>{0}</imgExt>\n'.format(seq.ext) + \
        '\t\t<startFrame>{0}</startFrame>\n'.format(seq.startFrame) + \
        '\t\t<endFrame>{0}</endFrame>\n'.format(seq.endFrame) + \
        '\t\t<writeImage>{0}</writeImage>\n'.format(0) + \
        '\t\t<showResult>{0}</showResult>\n'.format(0) + \
        '\t\t<outputFolder>{0}</outputFolder>\n'.format(path) + \
        '\t</sequence>\n' + \
        '</xml>\n'

    xmlfile.write(xmlstr)
    xmlfile.close()


    command = ['spt64.exe', name_xml]

    tic = time.clock()
    subprocess.call(command)
    duration = time.clock() - tic

    result = dict()
    res = np.loadtxt(path + '{0}.txt'.format(seq.name), dtype=int)
    result['res'] = res.tolist()
    result['type'] = 'rect'
    result['fps'] = round(seq.len / duration, 3)

    return result